### ISSUE TITLE
classify

### ISSUE THREAD

**Issue by paku55** [wnd-charm/wnd-charm#116]
I’m a PHD student. I would like to ask about “classify”. I have run the following command line in WNDCHRM 1.52-775. But, I don’t understand the generated output whether classify got successful or failure. I would appreciate it if you could tell some information.
> $ wndchrm classify -m -l -t1 -c st_54.fit /media/iSCSI-WNDCHRMDAT/YA/C3_20160610/a7_1/a7_1.tif

> Processing training set 'st_54.fit'.
----------
Summary of 'st_54.fit' (108 samples total, 1 samples per image):
'Class label' (interpreted value) number of samples.
'g3_54'	(0)	54
'n_54'	(0)	54
----------
Processing test set '/media/iSCSI-WNDCHRMDAT/YA/C3_20160610/a7_1/a7_1.tif'.
Adding '/media/iSCSI-WNDCHRMDAT/YA/C3_20160610/a7_1/a7_1-c-l.sig' for sig calc.
processing '/media/iSCSI-WNDCHRMDAT/YA/C3_20160610/a7_1/a7_1-c-l.sig' (index 0).
----------
Summary of '/media/iSCSI-WNDCHRMDAT/YA/C3_20160610/a7_1/a7_1.tif' (1 samples total, 1 samples per image):
Single class '' with 0 samples. Suitable as a test/classification set only.
1 unknown samples.
----------
Max balanced training images: 54
samples per image=1, training images: 54, testing images 0
----------
image	norm. fact.	p(g3_54)	p(n_54)	　act. class	pred. class	pred. val.
/media/iSCSI-WNDCHRMDAT/YA/C3_20160610/a7_1/a7_1.tif	6.01e+191.000	0.000	*	g3_54	0.000
----------
Average accuracy (1 splits): -nan
Confusion Matrix (sum of all splits)
                               g3_54              n_54      Total Tested  Per-Class Accuracy
             g3_54                 0                 0                 0             -nan
              n_54                 0                 0                 0             -nan
Average Similarity Matrix
                               g3_54              n_54
             g3_54           -nan           -nan
              n_54           -nan           -nan
Average Class Probability Matrix
                               g3_54              n_54
             g3_54           -nan           -nan
              n_54           -nan           -nan
----------
WARNING: Test set class label '' does not match any training set class.  Marked with '*'.

**Comment by igg** [comment]
The results are in a table with column headings starting with the line that begins with "image norm. fact. p(g3_54)...."
You only have one row, so it doesn't look like much of a table. Its tab-delimited, so you can copy/paste it into a spreadsheet along with the column headings.
The classifier is reporting a 100% probability that the unknown sample is g3_54

**Comment by paku55** [comment]
Thank you for your response. It helps me a lot. Classify output is different from wndchrm version.
Could you tell me some more questions? 
1. What does the following mean?  ”WARNING: Test set class label '' does not match any training set class. Marked with '*'.” 
2. Wndchrm computes its weighted distance in the training and classifies test images, so we shouldn’t confuse with stochastic concept such as probability of randomly selecting, I think. 
3. Image content descriptors are extracted the raw pixels and several transforms. Are Zernike Coefficients (), Zernike Coefficients (Fourier ())  and Zernike Coefficients (Chebyshev ()) entirely different features? Or same features group based on same description?
 I would appreciate it if you could tell again.

**Comment by igg** [comment]
> On Oct 12, 2017, at 08:34, paku55 <notifications@github.com> wrote:
> 
> Thank you for your response. It helps me a lot. Classify output is different from wndchrm version.
> Could you tell me some more questions?
> 
> What does the following mean? ”WARNING: Test set class label '' does not match any training set class. Marked with '*'.”

	Your blank test-class labels don't match any training class, so it will mark the true unknowns and not compute classifier performance statistics from them. Its just to let you know in case you thought you labeled your classes, that they were in fact  not labeled as far as wndchrm is concerned.

> Wndchrm computes its weighted distance in the training and classifies test images, so we shouldn’t confuse with stochastic concept such as probability of randomly selecting, I think.
I'm not really sure what you mean here. Its important to understand that by default the classifier is blinded to population structure by forcing evenly matched classes. So you have an equal chance of picking each class. The marginal probabilities that are reported are effectively a measure of similarity to each of the training classes.


> Image content descriptors are extracted the raw pixels and several transforms. Are Zernike Coefficients (), Zernike Coefficients (Fourier ()) and Zernike Coefficients (Chebyshev ()) entirely different features? Or same features group based on same description?
> I would appreciate it if you could tell again.

Zernike Coefficients are a single algorithm that operates on one greyscale pixel plane.  This pixel plane can either be the original image, or the original image transformed by one of the indicated transforms. A 2D-Fourier reads an input image and produces a 2D greyscale image as its output (ignoring the imaginary plane). Running the same feature extraction algorithm on an image vs. on a transform of the same image results in different image content descriptors, and hence different feature names, composed of the feature algorithm (Zernike) and any preceding image transform algorithms (Fourier, Chebyshev, etc).
